test/kernels/test_qkernel.py

```diff
@@ -515,5 +516,136 @@ def test_user_parameters(self):
 
 
+class TestQuantumKernelBatching(QiskitMachineLearningTestCase):
+    """Test QuantumKernel circuit batching
+
+    Creates a class factory that takes the QuantumInstance class,
+    applies a method decorator to the execute function and returns the modified class.
+    The decorator function records the actual number of circuits being passed
+    to execute does not exceed the batch_size requested by the Quantum Kernel.
+    Also checks that the sum of the batch sizes matches the total number of expected
+    circuits.
+    Checks batching with both statevector simulator and QASM simulator.
+    """
+
+    def count_circuits(self, func):
+        """Decorator to record the number of circuits passed to QuantumInstance.execute.
+
+        Args:
+            func (Callable): execute function to be wrapped
+
+        Returns:
+            Callable: function wrapper
+        """
+
+        @functools.wraps(func)
+        def wrapper(*args, **kwds):
+            self.circuit_counts.append(len(args[1]))
+            return func(*args, **kwds)
+
+        return wrapper
+
+    def hook_quantum_instance(self):
+        """Hooks into QuantumInstance to monitor circuit batching.
+
+        Class factory that takes the QuantumInstance class, applies a method decorator to the
+        execute function and returns the modified class.
+        """
+
+        class HookedQuantumInstance(QuantumInstance):
+            """Applies the count_circuits decorator to the QuantumInstance.execute function."""
+
+            @self.count_circuits
+            def execute(self, circuits, had_transpiled: bool = False):
+                return super().execute(circuits, had_transpiled)
+
+        return HookedQuantumInstance
+
+    def setUp(self):
+        super().setUp()
+
+        algorithm_globals.random_seed = 10598
+
+        self.shots = 12000
+        self.batch_size = 3
+        self.circuit_counts = []
+
+        self.statevector_simulator = self.hook_quantum_instance()(
+            BasicAer.get_backend("statevector_simulator"),
+            shots=1,
+            seed_simulator=algorithm_globals.random_seed,
+            seed_transpiler=algorithm_globals.random_seed,
+        )
+
+        self.qasm_simulator = self.hook_quantum_instance()(
+            BasicAer.get_backend("qasm_simulator"),
+            shots=self.shots,
+            seed_simulator=algorithm_globals.random_seed,
+            seed_transpiler=algorithm_globals.random_seed,
+        )
+
+        self.feature_map = ZZFeatureMap(feature_dimension=2, reps=2)
+
+        self.sample_train = np.asarray(
+            [
+                [3.07876080, 1.75929189],
+                [6.03185789, 5.27787566],
+                [6.22035345, 2.70176968],
+                [0.18849556, 2.82743339],
+            ]
+        )
+        self.label_train = np.asarray([0, 0, 1, 1])
+
+        self.sample_test = np.asarray([[2.199114860, 5.15221195], [0.50265482, 0.06283185]])
+        self.label_test = np.asarray([0, 1])
```

Comment:
Can you please add a line how these values were obtained? E.g. if I want to extend the dataset, how can I generate more samples.

Reply:
Sure, I've updated the data with a comment on its generation. Incidentally, I would probably have instinctively generated the data at runtime, but I note that data for unit tests is generally hardcoded throughout the module, presumably to avoid a dependency on ad_hoc_data.

Reply:
If test data is hard coded you definitely know it is not a random sample. Also, `ad_hoc` may be slow in some cases.